csharp-advanced-topics/Tests/Test.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text;
+using CsharpDelegatesActionFunc;
+using System;
+
+namespace Tests
+{
+    delegate string getInterestRate(double time, double rate);
+
+    [TestClass]
+    public class Test
+    {
+
+        [TestMethod]
+        public void whenRateAndTimeIsSentForSingleMethod_DelegateExecutesTheReferencedMethod()
+        {
+            getInterestRate delegateTotalInterest = new getInterestRate(InterestService.getTotalInterestByHours); 
+            string result = delegateTotalInterest.Invoke(8, 10);
+            Assert.AreEqual("Total Interest of $10 rate in 8 hours is 0.8", result);
+        }
```

Comment:
- Method names should be capitalized
- Please separate Arrange/Act/Assert with an empty line
- Use vars whenever possible, unless it's important to show which type it is or the compiler can't tell. (eg StringBuilder should be var)